### ISSUE TITLE
subproject default options cannot be overriden

### ISSUE THREAD

**Issue by inigomartinez** [mesonbuild/meson#2992]
[libgnome-volume-control](https://git.gnome.org/browse/libgnome-volume-control/) and [libgd](https://git.gnome.org/browse/libgd/) are using an user option called `static`, for building the library as `static` or `shared`, duplicating meson's [default_library](http://mesonbuild.com/Reference-manual.html#library) option.

However, there is no other option because they are expected to be configured as `subprojects`, but meson does not allow using [meson's options in subprojects](http://mesonbuild.com/Reference-manual.html#subproject): 
```
* `default_options` (added 0.37.0) an array of default option values that override those set in the subproject's `meson_options.txt` (like `default_options` in `project`, they only have effect when Meson is run for the first time, and command line arguments override any default options in build files)
```
This will cause confusion because the user will find two options where only one should be present and developers will expect to override them when using the `default_options` parameter in `subproject`.

**Comment by inigomartinez** [comment]
Thanks jibsen! However, this has arised when modifying [libgnome-volume-control](https://bugzilla.gnome.org/show_bug.cgi?id=792948#c10), which is already using a [workaround](https://git.gnome.org/browse/libgnome-volume-control/tree/meson_options.txt#n31).

I hope it gets some kind of improvement that solves this issue. I've been slightly looking at this and it might not be easy to fix, because the `default_library` option is part of the `Globals` variables that are shared between the parent and all the subprojects.

**Comment by kkartaltepe** [comment]
Tried to use `dependency(... static: true, fallback: ...)` as a workaround assuming that the static would be passed through to the fallback subproject but it is not. If it were I feel this would be a suitable alternative would it not? (as fallback is prefered over direct `subproject()` at least in the documentation).

**Comment by nirbheek** [comment]
This was fixed in https://github.com/mesonbuild/meson/pull/4157 for `default_library`. The fix is in 0.54
